### PR TITLE
Store hexadecimal values as binary

### PR DESCRIPTION
**Pull request by andbortnik**

### CODE REVIEW

File: thenewboston_node/business_logic/models/base.py [hunk 1]
```diff
@@ -43,6 +43,41 @@
...
+    return bytes.fromhex(value)
+
+
+def bytes_to_hex(value):
+    if value is None:
+        return value
+    return value.hex()
+
+
+COMPACT_VALUE_MAP = {
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/base.py:59]: also we use hexadecimals as keys here:
```
    accounts: dict[str, AccountBalance]
```

and here:

```
    updated_balances: dict[str, BlockAccountBalance] = {}

```

Please, take care of it too

File: thenewboston_node/business_logic/tests/test_models/test_field_compaction.py [hunk 2]
```diff
@@ -103,3 +103,102 @@ def test_can_load_block_from_compacted_dict():
     loaded_block = Block.from_compact_dict(compacted)
 
     assert loaded_block == block
+
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/tests/test_models/test_field_compaction.py:106]: add unittest the switching hexadecimals to binary result into reduced serialized size

**Review by dmugtasimov — CHANGES_REQUESTED** [review]

File: thenewboston_node/business_logic/tests/test_models/test_field_compaction.py (hunk 2 above)
Comment (andbortnik) [reply thenewboston_node/business_logic/tests/test_models/test_field_compaction.py:107]: https://github.com/thenewboston-developers/thenewboston-node/pull/114/files#diff-638e0785bcd0039bf74e9df592e40ef1fa8573cfd0afe997ff0a1aebbfe073eeR222

File: thenewboston_node/business_logic/models/base.py [hunk 3]
```diff
@@ -43,6 +43,41 @@
...
+    'node_identifier': bytes_to_hex,
+    'message_hash': bytes_to_hex,
+    'message_signature': bytes_to_hex,
+    'block_identifier': bytes_to_hex,
+    'balance_lock': bytes_to_hex,
+    'recipient': bytes_to_hex,
+    'sender': bytes_to_hex,
+    'lock': bytes_to_hex,
+}
+
```
[the remark this answers is not in the document]
Comment (andbortnik) [reply thenewboston_node/business_logic/models/base.py:80]: https://github.com/thenewboston-developers/thenewboston-node/pull/114/files#diff-b89d0377b3dd52d8434eb93fabbfdd534f2927013297055cbab7735ae9eb38c0R93

File: thenewboston_node/business_logic/models/base.py (hunk 1 above)
Comment (andbortnik) [reply thenewboston_node/business_logic/models/base.py:59]: Done

File: thenewboston_node/business_logic/models/base.py [hunk 4]
```diff
@@ -43,10 +43,57 @@
...
+UNCOMPACT_SUBKEY_MAP = {
+    'updated_balances': bytes_to_hex,
+    'accounts': bytes_to_hex,
+}
+
 # Assert that bidirectional mapping is defined correctly
 assert len(COMPACT_KEY_MAP) == len(UNCOMPACT_KEY_MAP)
 assert COMPACT_KEY_MAP.keys() == set(UNCOMPACT_KEY_MAP.values())
 assert UNCOMPACT_KEY_MAP.keys() == set(COMPACT_KEY_MAP.values())
+assert set(COMPACT_VALUE_MAP.keys()) == set(UNCOMPACT_VALUE_MAP.keys())
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/base.py:95]: keys() should already be a set in Python 3

**Review by dmugtasimov — APPROVED** [review]

**Pull request merged**